pkg/client/apiutil/restmapper.go

```diff
@@ -194,13 +214,26 @@ func (m *mapper) addKnownGroupAndReload(groupName string, versions ...string) er
 		return fmt.Errorf("failed to get API group resources: %w", err)
 	}
 
-	if _, ok := m.knownGroups[groupName]; ok {
-		groupResources = m.knownGroups[groupName]
-	}
+	m.addGroupVersionResourcesToCacheLocked(groupVersionResources)
+	return nil
+}
 
+// addGroupVersionResourcesToCacheLocked does what the name suggests. The mutex must be held when
+// calling it.
+func (m *mapper) addGroupVersionResourcesToCacheLocked(gvr map[schema.GroupVersion]*metav1.APIResourceList) {
```

Comment:
I know the name of this func is getting long, but might be still good to mention that it also reloads the mapper, maybe `addGroupVersionResourcesToCacheAndReloadLocked` (would align to `addKnownGroupAndReload`)

Reply:
done